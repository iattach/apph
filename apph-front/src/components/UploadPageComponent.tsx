export default function UploadPageComponent() {
  return <div>Page d'upload</div>;
}
