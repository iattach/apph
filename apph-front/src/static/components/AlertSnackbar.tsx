import { AlertColor, Snackbar } from '@mui/material';
import React from 'react';
import MuiAlert, { AlertProps } from '@mui/material/Alert';

const Alert = React.forwardRef<HTMLDivElement, AlertProps>(function Alert(
  props,
  ref
) {
  return <MuiAlert elevation={6} ref={ref} variant="filled" {...props} />;
});

type Props = {
  open: boolean;
  severity?: AlertColor;
  message: string;
  onClose: (b: boolean) => void;
};

export const AlertSnackbar = ({ open, severity, message, onClose }: Props) => {
  const handleSnackbarClose = (
    event?: React.SyntheticEvent | Event,
    reason?: string
  ) => {
    if (reason === 'clickaway') return;
    onClose(false);
  };

  return (
    <Snackbar
      open={open}
      autoHideDuration={6000}
      anchorOrigin={{ vertical: 'bottom', horizontal: 'center' }}
      onClose={handleSnackbarClose}
    >
      <Alert onClose={handleSnackbarClose} severity={severity}>
        {message}
      </Alert>
    </Snackbar>
  );
};
