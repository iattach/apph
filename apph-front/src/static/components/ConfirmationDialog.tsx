import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle
} from '@mui/material';

type Props = {
  open: boolean;
  message?: string;
  title?: string;
  onConfirm?: () => void;
  onCancel?: () => void;
};
export const ConfirmationDialog = ({
  open,
  message,
  title,
  onConfirm,
  onCancel
}: Props) => {
  return (
    <Dialog open={open} sx={{ py: '20vh' }}>
      <DialogTitle>{title}</DialogTitle>
      <DialogContent>
        <DialogContentText>{message}</DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button color="primary" onClick={onConfirm}>
          Continuer
        </Button>
        <Button color="error" onClick={onCancel}>
          Annuler
        </Button>
      </DialogActions>
    </Dialog>
  );
};
