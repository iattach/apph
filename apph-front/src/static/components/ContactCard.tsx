import { Card, CardHeader, IconButton, Tooltip } from '@mui/material';
import * as React from 'react';
import { UserAvatar } from './UserAvatar';
import { Delete } from '@mui/icons-material';
import { useTranslation } from 'react-i18next';

export const ContactCard = ({
  firstname,
  lastname,
  login
}: {
  firstname: string;
  lastname: string;
  login: string;
}): JSX.Element => {
  const { t } = useTranslation();

  return (
    <Card>
      <CardHeader
        title={`${firstname} ${lastname}`}
        subheader={login}
        avatar={<UserAvatar firstname={firstname} lastname={lastname} />}
        action={
          <Tooltip title={t('action.delete')}>
            <IconButton>
              <Delete />
            </IconButton>
          </Tooltip>
        }
      />
    </Card>
  );
};
