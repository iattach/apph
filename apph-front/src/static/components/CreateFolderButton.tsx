import Button from '@mui/material/Button';
import React, { Dispatch, SetStateAction, useState } from 'react';
import {
  CircularProgress,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  TextField
} from '@mui/material';
import { FolderService } from '../../services/FolderService';
import { IFolder } from '../../utils';
import { useTranslation } from 'react-i18next';

export const CreateFolderButton = ({
  selected,
  setRootFolder
}: {
  selected: string;
  setRootFolder: Dispatch<SetStateAction<IFolder | null>>;
}): JSX.Element => {
  const [showModal, setShowModal] = useState<boolean>(false);
  const [folderName, setFolderName] = useState<string>('');
  const [errorMessage, setErrorMessage] = useState('');
  const [loading, setLoading] = useState<boolean>(false);
  const { t } = useTranslation();
  const handleSubmit = async () => {
    if (folderName === '') {
      setErrorMessage('folder.error.emptyFolder');
    } else {
      setLoading(true);
      await FolderService.createFolder(
        folderName,
        selected,
        (folder: IFolder) => {
          setRootFolder(folder);
          setShowModal(false);
        },
        (error: string) => {
          setErrorMessage(error);
        }
      );
      setLoading(false);
    }
  };

  return (
    <>
      <Button
        onClick={() => {
          setFolderName('');
          setShowModal(true);
        }}
      >
        {t('folder.createFolder')}
      </Button>
      <Dialog
        open={showModal}
        onClose={() => {
          if (!loading) {
            setShowModal(false);
          }
        }}
      >
        <DialogTitle sx={{ fontWeight: 'bold' }}>
          {t('folder.creation')}
        </DialogTitle>
        <DialogContent>
          <DialogContentText sx={{ color: 'black' }}>
            {t('folder.enterName')}
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            value={folderName}
            onChange={(event) => {
              setFolderName(event.currentTarget.value);
            }}
            id="folder-name"
            autoComplete={t('folder.new')}
            label={t('folder.name')}
            size="small"
            disabled={loading}
            inputProps={{ maxLength: 255 }}
            onKeyDown={async (event) => {
              if (event.key === 'Enter') {
                event.preventDefault();
                await handleSubmit();
              }
            }}
          />
          <DialogContentText
            sx={{ color: 'red', fontSize: 'small' }}
            hidden={!errorMessage}
          >
            {t(errorMessage)}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button
            onClick={async (event) => {
              event.preventDefault();
              await handleSubmit();
            }}
            disabled={loading}
          >
            {loading ? <CircularProgress /> : <>{t('folder.create')}</>}
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};
