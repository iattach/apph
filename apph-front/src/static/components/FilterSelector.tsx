import * as React from 'react';
import { useEffect, useReducer } from 'react';
import { AddCircle } from '@mui/icons-material';
import { FilterComponent } from './FilterComponent';
import { Box, IconButton, Tooltip } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import TagService from '../../services/TagService';
import { ITag } from '../../utils';
import { setTagList } from '../../redux/slices/tagSlice';
import { useDispatch, useSelector } from 'react-redux';
import { IFilter } from '../../utils/types/Filter';

const filterSelectorBoxStyle = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  '> div, button': { mt: '1vh' }
};

const filterButtonStyle = { mb: '2vh', mx: '1vh' };

const isInstanceOfIFilter = (object: object): object is IFilter => {
  return (
    'id' in object &&
    ('field' in object || 'operator' in object || 'value' in object)
  );
};

export enum filterActionKind {
  ADD = 'addFilter',
  REMOVE = 'removeFilter',
  UPDATE = 'updateFilter'
}

export interface filterActions {
  type: filterActionKind;
  payload: IFilter | number;
}

const filterInitialState: IFilter[] = [
  {
    id: 0,
    field: '',
    operator: '',
    value: ''
  }
];

const filterReducers = (state: IFilter[], action: filterActions) => {
  const { type, payload } = action;
  switch (type) {
    case filterActionKind.ADD:
      if (typeof payload == 'number')
        return [
          ...state,
          {
            id: payload,
            field: '',
            operator: '',
            value: ''
          }
        ];
      throw new Error(
        "Erreur dans l'ajout du filtre: le payload n'est pas un nombre."
      );
    case filterActionKind.REMOVE:
      if (typeof payload == 'number')
        return state.filter((filter) => filter.id != action.payload);
      throw new Error(
        "Erreur dans la suppression du filtre: le payload n'est pas un nombre"
      );
    case filterActionKind.UPDATE:
      if (typeof payload == 'object' && isInstanceOfIFilter(payload)) {
        const filterIndex = state.findIndex(
          (filter) => filter.id == payload.id
        );
        state[filterIndex] = {
          id: state[filterIndex].id,
          field:
            payload.field !== 'undefined'
              ? payload.field
              : state[filterIndex].field,
          operator:
            payload.field === 'tags'
              ? null
              : payload.operator !== 'undefined'
              ? payload.operator
              : state[filterIndex].operator,
          value:
            payload.value !== 'undefined'
              ? payload.value
              : state[filterIndex].value
        };
        return [...state];
      }
      throw new Error(
        "Erreur dans la mise Ã  jour du filtre : l'id ne correspond Ã  aucun filtre existant"
      );
    default:
      throw new Error('Erreur dans la gestion du filtre');
  }
};

function* generateId() {
  let id = 1;
  while (true) yield id++;
}
const iterator = generateId();

interface filterSelectorProps {
  onFilterPhoto: (filterList: IFilter[]) => void;
  openAlert: (message: string) => void;
}

export const FilterSelector = ({
  onFilterPhoto,
  openAlert
}: filterSelectorProps) => {
  const [filterStates, dispatchFilterStates] = useReducer(
    filterReducers,
    filterInitialState
  );
  const tagList = useSelector(({ tagList }: { tagList: ITag[] }) => tagList);
  const dispatch = useDispatch();

  useEffect(() => {
    TagService.getAllTags(
      (tags: string) => {
        const tagsConverted: ITag[] = JSON.parse(tags);
        dispatch(setTagList(tagsConverted));
      },
      (errorMessage: string) => openAlert(errorMessage)
    );
  }, []);

  const createNewFilter = () => {
    const currentIterator = iterator.next();
    if (!currentIterator.done)
      dispatchFilterStates({
        type: filterActionKind.ADD,
        payload: currentIterator.value
      });
  };

  const handleFilterPhoto = () => {
    onFilterPhoto(filterStates);
  };

  const getFilterList = () =>
    filterStates.map((filterState) => (
      <FilterComponent
        key={filterState.id}
        state={filterState}
        dispatchFilterState={dispatchFilterStates}
        tagList={tagList}
      />
    ));

  return (
    <Box sx={filterSelectorBoxStyle}>
      {getFilterList()}
      <Box>
        <Tooltip title="Ajouter un filtre" arrow>
          <IconButton onClick={createNewFilter} sx={filterButtonStyle}>
            <AddCircle />
          </IconButton>
        </Tooltip>
        <Tooltip title="Rechercher" arrow>
          <IconButton
            color="primary"
            onClick={handleFilterPhoto}
            sx={filterButtonStyle}
          >
            <SearchIcon />
          </IconButton>
        </Tooltip>
      </Box>
    </Box>
  );
};
