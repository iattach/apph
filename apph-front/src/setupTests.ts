// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import { ReactElement } from 'react';
import { wrapper } from './tests/utils/components/CustomWrapper';
import { render } from '@testing-library/react';

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: jest.fn(),
        language: 'fr'
      }
    };
  }
}));

jest.mock('ts-debounce', () => ({
  debounce: (func: (query: string) => Promise<void>, waitMs: number) => func
}));

const mockedUseNavigate = jest.fn();
const mockedUseLocation = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
  useLocation: () => mockedUseLocation
}));

const renderWithWrapper = (component: ReactElement) =>
  render(component, { wrapper });

export { mockedUseNavigate, mockedUseLocation, renderWithWrapper };
