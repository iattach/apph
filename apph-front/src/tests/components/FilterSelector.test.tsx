import { screen } from '@testing-library/react';
import { clickButton, fillText, selectOptionInListBox } from '../utils';
import { FilterSelector } from '../../static/components/FilterSelector';
import { renderWithWrapper } from '../../setupTests';

describe('Test du fonctionnement des filtres', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('test if a filter is correctly added', () => {
    //GIVEN
    const onError = () => null;
    const onFilterPhoto = () => null;
    renderWithWrapper(
      <FilterSelector onError={onError} onFilterPhoto={onFilterPhoto} />
    );
    //WHEN
    clickButton(/filter.add/);
    //THEN
    expect(screen.getAllByLabelText(/filter.field/).length).toBe(2);
  });

  it('test if a filter is correctly deleted', () => {
    //GIVEN
    const onError = () => null;
    const onFilterPhoto = () => null;
    renderWithWrapper(
      <FilterSelector onError={onError} onFilterPhoto={onFilterPhoto} />
    );
    //WHEN
    clickButton(/filter.delete/);
    //THEN
    expect(screen.queryByText(/filter.field/)).not.toBeInTheDocument();
  });

  it('test that filter datas are correctly submitted', () => {
    //GIVEN
    const spyFilterPhoto = jest.fn();
    const spyOpenAlert = jest.fn();
    renderWithWrapper(
      <FilterSelector onFilterPhoto={spyFilterPhoto} onError={spyOpenAlert} />
    );
    const fieldInput = screen.getByLabelText(/filter.field/);
    const operatorInput = screen.getByLabelText(/filter.operator/);
    //WHEN
    selectOptionInListBox(fieldInput, /photoTable.title/);
    selectOptionInListBox(operatorInput, /operator.equal/);
    fillText(/filter.value/, 'Test');
    clickButton(/action.search/);
    //THEN
    expect(spyFilterPhoto).toBeCalledWith([
      {
        id: 0,
        field: 'title',
        operator: 'is',
        value: 'Test'
      }
    ]);
  });

  it('should render with tagName', () => {
    const spyFilterPhoto = jest.fn();
    const spyOpenAlert = jest.fn();

    const tree = renderWithWrapper(
      <FilterSelector
        tagName={'Tag_1'}
        onFilterPhoto={spyFilterPhoto}
        onError={spyOpenAlert}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
