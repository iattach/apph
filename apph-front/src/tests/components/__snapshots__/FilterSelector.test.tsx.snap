// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test du fonctionnement des filtres should render with tagName 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="MuiBox-root css-kzat4l"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="select-field-label"
            >
              filter.field
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1xsr2on-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="select-field-label select-field"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="select-field"
                role="button"
                tabindex="0"
              >
                photoTable.title
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="title"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="select-operator-label"
            >
              filter.operator
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1ckzzvk-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="select-operator-label select-operator"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="select-operator"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="fill-value"
                id="fill-value-label"
              >
                filter.value
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  disabled=""
                  id="fill-value"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1ftyaf0"
                  >
                    <span>
                      filter.value
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <button
            aria-label="filter.delete"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RemoveCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            aria-label="filter.add"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-y81peu-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddCircleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="action.search"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-celzet-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-kzat4l"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            id="select-field-label"
          >
            filter.field
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1xsr2on-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select-field-label select-field"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="select-field"
              role="button"
              tabindex="0"
            >
              photoTable.title
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="title"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="select-operator-label"
          >
            filter.operator
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1ckzzvk-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select-operator-label select-operator"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="select-operator"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="fill-value"
              id="fill-value-label"
            >
              filter.value
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled=""
                id="fill-value"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    filter.value
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          aria-label="filter.delete"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RemoveCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11H7v-2h10v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <button
          aria-label="filter.add"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-y81peu-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddCircleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="action.search"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-celzet-MuiButtonBase-root-MuiIconButton-root"
          data-mui-internal-clone-element="true"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
